File: solutions/nim/sum-of-multiples/1/sum_of_multiples.nim
import std/sets

proc sum*(limit: int, factors: openArray[int]): int =
    var uniqueMultiples = initHashSet[int]()
    for factor in factors:
        if factor == 0: continue
        var multiple = factor
        while multiple < limit:
            uniqueMultiples.incl(multiple)
            multiple += factor

    for multiple in uniqueMultiples:
        result += multiple

File: solutions/python/sum-of-multiples/1/sum_of_multiples.py
"""Sum of Multiples exercise on Exercism's Python Track"""

from itertools import chain

def sum_of_multiples(limit, factors):
    """Add up all the unique multiples of the factors that are less than the limit.
    :param limit: int - upper limit of the multiples
    :param factors: list[int] - list of factors
    :return: int - the resulting sum
    """

    nonzero_factors = filter(lambda f: f != 0, factors)
    factor_multiples = map(lambda factor: range(factor, limit, factor), nonzero_factors)
    all_multiples = chain(*factor_multiples)
    unique_multiples = set(all_multiples)

    return sum(unique_multiples)
